internal/services/artifact.go

```diff
@@ -74,3 +80,82 @@ func (s *artifactService) GetArtifactPolicies(ctx context.Context, artifact stri
 		},
 	}, nil
 }
+
+func (s *artifactService) GetImageMetadata(ctx context.Context, image string, username string, password string) (models.ImageMetadataResponse, error) {
+	ref, err := name.ParseReference(image)
+	if err != nil {
+		return models.ImageMetadataResponse{}, fmt.Errorf("invalid image URI: %w", err)
+	}
+
+	auth := authn.Anonymous
+	if username != "" && password != "" {
+		auth = &authn.Basic{Username: username, Password: password}
+	}
+
+	opts := []remote.Option{remote.WithAuth(auth), remote.WithContext(ctx)}
+
+	descriptor, err := remote.Get(ref, opts...)
+	if err != nil {
+		if isNotFound(err) {
+			return models.ImageMetadataResponse{}, fmt.Errorf("Image not found: %w", err)
+		} else if isAuthError(err) {
+			return models.ImageMetadataResponse{}, fmt.Errorf("Authentication failed: %w", err)
+		} else {
+			return models.ImageMetadataResponse{}, fmt.Errorf("Failed to fetch metadata: %w", err)
+		}
+	}
+
+	// Fetch digest
+	img, err := remote.Image(ref, opts...)
+	if err != nil {
+		return models.ImageMetadataResponse{}, fmt.Errorf("Failed to fetch image: %w", err)
+	}
+	digest, err := img.Digest()
+	if err != nil {
+		return models.ImageMetadataResponse{}, fmt.Errorf("Failed to compute digest: %w", err)
+	}
+
+	// Extract config metadata
+	configFile, err := img.ConfigFile()
+	if err != nil {
+		return models.ImageMetadataResponse{}, fmt.Errorf("Failed to fetch config file: %w", err)
+	}
+
+	created := configFile.Created
+	labels := configFile.Config.Labels
+	if len(labels) == 0 {
+		labels = nil
+	}
+
+	response := models.ImageMetadataResponse{
+		Image: &image,
+		Metadata: models.Metadata{
+			MediaType: string(descriptor.MediaType),
+			Size:      descriptor.Size,
+			Created:   &created.Time,
+			Labels:    &labels,
+		},
+		Digest: digest.String(),
+	}
+	return response, nil
+}
+
+// isNotFound checks if the error indicates the image was not found
+func isNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(strings.ToLower(err.Error()), "not found") ||
+		strings.Contains(strings.ToLower(err.Error()), "404") ||
```

Comment:
If you switch to a dedicated errors package you can use `errors.Is` as opposed to strings.Contains in places :)